app/src/test/java/com/codesoom/assignment/service/ProductServiceTest.java

```diff
@@ -0,0 +1,116 @@
+package com.codesoom.assignment.service;
+
+import com.codesoom.assignment.domain.Product;
+import com.codesoom.assignment.domain.ProductJpaRepository;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@SpringBootTest()
+@RunWith(SpringRunner.class)
+@Transactional
+class ProductServiceTest {
+
+    @Autowired
+    private ProductService productService;
+
+
+    @DisplayName("findAll method")
+    @Nested
+    class Describe_findAll {
+
+
+        @DisplayName("if nothing is added")
+        @Nested
+        class Context_no_product {
+
+            @BeforeEach
+            void setup() {
+                productService.deleteAll();
+
+            }
+            @DisplayName("returns an empty list ")
+            @Test
+            void it_returns_empty() {
+                List<Product> all = productService.findAll();
+                assertThat(all.size()).isEqualTo(0);
+            }
+        }
+
+        @DisplayName("if two products are added")
+        @Nested
+        class Context_two_products {
+
+            @BeforeEach
+            void setup() {
+                productService.deleteAll();
+            }
+
+            @DisplayName("returns a list of two products ")
+            @Test
+            void it_returns_empty() {
+                productService.save(Product.builder().name("name").maker("maker").imageUrl("url").build());
+                productService.save(Product.builder().name("name2").maker("maker2").imageUrl("url2").build());
+
+                List<Product> all = productService.findAll();
+                assertThat(all.size()).isEqualTo(2);
+            }
+        }
+    }
+
+    @DisplayName("save method")
+    @Nested
+    class Describe_save {
+        @After
+        void after() {
+            productService.deleteAll();
+        }
+
+        @DisplayName("if one product is saved")
+        @Nested
+        class Context_one_product {
+            private Product product = Product.builder().name("name").maker("maker").imageUrl("url").price(1000L).build();
```

Comment:
한 줄이 너무 길군요. 한 줄에 하나의 의도만 남도록 한 줄에 `.`이 하나씩 오도록 개행하는 습관을 가져 봅시다.

```suggestion
            private Product product = Product.builder()
              .name("name")
              .maker("maker")
              .imageUrl("url")
              .price(1000L)
              .build();
```

Reply:
개행했습니다!